pkg/common/log/logger.go

```diff
@@ -120,12 +133,17 @@ func (l *MOLogger) Log(msg string, opts LogOptions, fields ...zap.Field) bool {
 		}
 	}
 
-	if ce := l.logger.Check(opts.level, msg); ce != nil {
+	if opts.ctx == nil {
+		opts.ctx = l.ctx
+	}
+
+	logger := l.logger.WithOptions(zap.AddCallerSkip(opts.callerSkip + 1))
```

Comment:
Writing this way will have memory allocation, find ways to avoid it as much as possible as logging can be very frequent

Reply:
use inited map in MOLogger like
```
MOLogger {
  m:  map[int]*zap.Logger {
     1: logger.WithOption(zap.AddCallerSkip(1)),
     2: logger.WithOption(zap.AddCallerSkip(2)),
     3: logger.WithOption(zap.AddCallerSkip(3)),
  }
}
```